File: API/Core/Entities/Invoice.cs
using System;
using System.Collections.ObjectModel;

namespace API.Core.Entities
{
    public class Invoice
    {
        public int Id { get; set; }
        public int CustomerId { get; set; }
        public string Title { get; set; }
        public string InvoiceAddress { get; set; }
        public DateTime InvoiceDate { get; set; } = DateTime.UtcNow;
        public bool InvoiceYesNo { get; set; } = false;
        public decimal TotalValue { get; set; } = 0.0M;
        public decimal TotalPaid { get; set; } = 0M;
        public DateTime? DatePaid { get; set; }
        public bool IsPosted { get; set; } = false;
        public string InvoiceNote { get; set; }
        public Customer Customer { get; set; }
        public Collection<InvoiceItem> InvoiceItems { get; set; }
    }
}

File: client/src/app/shared/Models/Invoice.ts
import { InvoiceItem } from './InvoiceItem';

export interface IInvoice {
    id: number;
    customerId: number;
    title: string;
    invoiceAddress: string;
    invoiceDate: Date;
    invoiceYesNo: boolean;
    totalValue: number;
    totalPaid: number;
    datePaid: Date;
    isPosted: boolean;
}

export class Invoice {
    id: number;
    customerId: number;
    title: string;
    invoiceAddress: string;
    invoiceDate: Date;
    invoiceYesNo: boolean;
    totalValue: number;
    totalPaid: number;
    datePaid: Date;
    isPosted: boolean;
}
export class InvoiceToInsert {
    id: number;
    customerId: number;
    invoiceAddress: string;
    title: string;
    invoiceDate: Date;
    invoiceYesNo: boolean;
    totalValue: number;
    totalPaid: number;
    datePaid: Date;
    isPosted: boolean;
    invoiceItems: InvoiceItem[] = [];
}
